--- a/net/minecraft/world/level/storage/loot/entries/NestedLootTable.java
+++ b/net/minecraft/world/level/storage/loot/entries/NestedLootTable.java
@@ -29,11 +_,11 @@
             return "->{inline}";
         }
     };
-    private final Either<ResourceKey<LootTable>, LootTable> contents;
+    public final Either<ResourceKey<LootTable>, LootTable> contents; // Purpur - private -> public (Jade protocol)
 
     private NestedLootTable(
         final Either<ResourceKey<LootTable>, LootTable> contents,
         final int weight,
         final int quality,
         final List<LootItemCondition> conditions,
         final List<LootItemFunction> functions
