--- a/net/minecraft/world/level/block/entity/BarrelBlockEntity.java
+++ b/net/minecraft/world/level/block/entity/BarrelBlockEntity.java
@@ -89,6 +_,16 @@
         this.items = items;
     }
 
+    // Purpur start - pca
+    @Override
+    public void setChanged() {
+        super.setChanged();
+        if (org.purpurmc.purpur.PurpurConfig.protocol.pca.enable) {
+            org.purpurmc.purpur.protocol.PcaSyncProtocol.syncBlockEntityToClient(this);
+        }
+    }
+    // Purpur end - pca
+
     @Override
     protected Component getDefaultName() {
         return DEFAULT_NAME;
