--- a/net/minecraft/world/level/block/entity/BeehiveBlockEntity.java
+++ b/net/minecraft/world/level/block/entity/BeehiveBlockEntity.java
@@ -80,7 +_,7 @@
         "leash",
         "UUID"
     );
-    public static final int MAX_OCCUPANTS = 3;
+    public static final int MAX_OCCUPANTS = org.purpurmc.purpur.PurpurConfig.beeInsideBeeHive; // Purpur - Config to change max number of bees
     private static final int MIN_TICKS_BEFORE_REENTERING_HIVE = 400;
     private static final int MIN_OCCUPATION_TICKS_NECTAR = 2400;
     public static final int MIN_OCCUPATION_TICKS_NECTARLESS = 600;
@@ -150,5 +_,10 @@
             super.setChanged();
         }
 
+        // Purpur start - pca
+        if (org.purpurmc.purpur.PurpurConfig.protocol.pca.enable) {
+            org.purpurmc.purpur.protocol.PcaSyncProtocol.syncBlockEntityToClient(this);
+        }
+        // Purpur end - pca
         return spawned;
     }
@@ -157,10 +_,32 @@
         return spawned;
     }
 
+    // Purpur start - Stored Bee API
+    public List<Entity> releaseBee(BlockState iblockdata, BeehiveBlockEntity.BeeData data, BeehiveBlockEntity.BeeReleaseStatus tileentitybeehive_releasestatus, boolean force) {
+        List<Entity> list = Lists.newArrayList();
+
+        BeehiveBlockEntity.releaseOccupant(this.level, this.worldPosition, iblockdata, data.occupant, list, tileentitybeehive_releasestatus, this.savedFlowerPos, force);
+
+        if (!list.isEmpty()) {
+            stored.remove(data);
+
+            super.setChanged();
+        }
+
+        return list;
+    }
+    // Purpur end - Stored Bee API
+
     @VisibleForDebug
     public int getOccupantCount() {
         return this.stored.size();
     }
+
+    // Purpur start - Stored Bee API
+    public List<BeeData> getStored() {
+        return stored;
+    }
+    // Purpur end - Stored Bee API
 
     // Paper start - Add EntityBlockStorage clearEntities
     public void clearBees() {
@@ -207,6 +_,11 @@
                 this.level.gameEvent(GameEvent.BLOCK_CHANGE, blockPos, GameEvent.Context.of(bee, this.getBlockState()));
             }
 
+            // Purpur start - pca
+            if (org.purpurmc.purpur.PurpurConfig.protocol.pca.enable) {
+                org.purpurmc.purpur.protocol.PcaSyncProtocol.syncBlockEntityToClient(this);
+            }
+            // Purpur end - pca
             bee.discard(org.bukkit.event.entity.EntityRemoveEvent.Cause.ENTER_BLOCK); // CraftBukkit - add Bukkit remove cause
             super.setChanged();
         }
@@ -328,5 +_,15 @@
         this.stored
             .removeIf(
-                occupantEntry -> releaseOccupant(this.level, this.worldPosition, state, occupantEntry.toOccupant(), spawned, releaseStatus, this.savedFlowerPos, force) // CraftBukkit - This allows us to bypass the night/rain/emergency check
+                occupantEntry -> {
+                    if (releaseOccupant(this.level, this.worldPosition, state, occupantEntry.toOccupant(), spawned, releaseStatus, this.savedFlowerPos, force)) {
+                        // Purpur start - pca
+                        if (org.purpurmc.purpur.PurpurConfig.protocol.pca.enable) {
+                            org.purpurmc.purpur.protocol.PcaSyncProtocol.syncBlockEntityToClient(java.util.Objects.requireNonNull(this.level.getBlockEntity(this.worldPosition)));
+                        }
+                        // Purpur end - pca
+                        return true;
+                    }
+                    return false;
+                }
             );
         if (!spawned.isEmpty()) {
@@ -359,5 +_,10 @@
         this.stored = Lists.newArrayList(); // CraftBukkit - SPIGOT-7790: create new copy (may be modified in physics event triggered by honey change)
         input.read("bees", BeehiveBlockEntity.Occupant.LIST_CODEC).orElse(List.of()).forEach(this::storeBee);
         this.savedFlowerPos = input.read("flower_pos", BlockPos.CODEC).orElse(null);
         this.maxBees = input.getIntOr("Bukkit.MaxEntities", MAX_OCCUPANTS); // Paper - persist max bukkit occupants
+        // Purpur start - pca
+        if (org.purpurmc.purpur.PurpurConfig.protocol.pca.enable) {
+            org.purpurmc.purpur.protocol.PcaSyncProtocol.syncBlockEntityToClient(this);
+        }
+        // Purpur end - pca
     }
