--- a/net/minecraft/world/level/block/entity/ChestBlockEntity.java
+++ b/net/minecraft/world/level/block/entity/ChestBlockEntity.java
@@ -181,6 +_,16 @@
         two.setItems(items);
     }
 
+    // Purpur start - pca
+    @Override
+    public void setChanged() {
+        super.setChanged();
+        if (org.purpurmc.purpur.PurpurConfig.protocol.pca.enable) {
+            org.purpurmc.purpur.protocol.PcaSyncProtocol.syncBlockEntityToClient(this);
+        }
+    }
+    // Purpur end - pca
+
     @Override
     protected AbstractContainerMenu createMenu(final int containerId, final Inventory inventory) {
         return ChestMenu.threeRows(containerId, inventory, this);
