--- a/net/minecraft/world/level/block/entity/HopperBlockEntity.java
+++ b/net/minecraft/world/level/block/entity/HopperBlockEntity.java
@@ -89,6 +_,16 @@
         this.facing = blockState.getValue(HopperBlock.FACING);
     }
 
+    // Purpur start - pca
+    @Override
+    public void setChanged() {
+        super.setChanged();
+        if (org.purpurmc.purpur.PurpurConfig.protocol.pca.enable) {
+            org.purpurmc.purpur.protocol.PcaSyncProtocol.syncBlockEntityToClient(this);
+        }
+    }
+    // Purpur end - pca
+
     @Override
     protected Component getDefaultName() {
         return DEFAULT_NAME;
@@ -200,5 +_,10 @@
             if (changed) {
                 entity.setCooldown(level.spigotConfig.hopperTransfer); // Spigot
                 setChanged(level, pos, state);
+                // Purpur start - pca
+                if (org.purpurmc.purpur.PurpurConfig.protocol.pca.enable) {
+                    org.purpurmc.purpur.protocol.PcaSyncProtocol.syncBlockEntityToClient(entity);
+                }
+                // Purpur end - pca
                 return true;
             }
