--- a/net/minecraft/world/entity/animal/equine/AbstractHorse.java
+++ b/net/minecraft/world/entity/animal/equine/AbstractHorse.java
@@ -300,5 +_,12 @@
     public void createInventory() {
         SimpleContainer old = this.inventory;
         this.inventory = new SimpleContainer(this.getInventorySize(), (org.bukkit.entity.AbstractHorse) this.getBukkitEntity()); // CraftBukkit
+        // Purpur start - pca
+        this.inventory.addListener(inv -> {
+            if (org.purpurmc.purpur.PurpurConfig.protocol.pca.enable) {
+                org.purpurmc.purpur.protocol.PcaSyncProtocol.syncEntityToClient(this);
+            }
+        });
+        // Purpur end - pca
         if (old != null) {
             int max = Math.min(old.getContainerSize(), this.inventory.getContainerSize());
