--- a/net/minecraft/world/entity/animal/equine/AbstractHorse.java
+++ b/net/minecraft/world/entity/animal/equine/AbstractHorse.java
@@ -300,4 +_,12 @@
     public void createInventory() {
         SimpleContainer old = this.inventory;
-        this.inventory = new SimpleContainer(this.getInventorySize(), (org.bukkit.entity.AbstractHorse) this.getBukkitEntity()); // CraftBukkit
+        this.inventory = new SimpleContainer(this.getInventorySize(), (org.bukkit.entity.AbstractHorse) this.getBukkitEntity()) { // CraftBukkit // Purpur - pca
+            @Override
+            public void setChanged() {
+                super.setChanged();
+                if (org.purpurmc.purpur.PurpurConfig.protocol.pca.enable) {
+                    org.purpurmc.purpur.protocol.PcaSyncProtocol.syncEntityToClient(AbstractHorse.this);
+                }
+            }
+        }; // Purpur - pca
         if (old != null) {
