--- a/net/minecraft/world/entity/npc/villager/AbstractVillager.java
+++ b/net/minecraft/world/entity/npc/villager/AbstractVillager.java
@@ -49,6 +_,15 @@
         super(type, level);
         this.setPathfindingMalus(PathType.DANGER_FIRE, 16.0F);
         this.setPathfindingMalus(PathType.DAMAGE_FIRE, -1.0F);
+        // Purpur start - pca
+        if (!this.level().isClientSide()) {
+            this.inventory.addListener(inventory -> {
+                if (org.purpurmc.purpur.PurpurConfig.protocol.pca.enable) {
+                    org.purpurmc.purpur.protocol.PcaSyncProtocol.syncEntityToClient(this);
+                }
+            });
+        }
+        // Purpur end - pca
     }
 
     @Override
