--- a/net/minecraft/network/protocol/common/custom/CustomPacketPayload.java
+++ b/net/minecraft/network/protocol/common/custom/CustomPacketPayload.java
@@ -41,3 +_,7 @@
             public void encode(final B output, final CustomPacketPayload value) {
+                // Purpur start - protocol core
+                if (value instanceof org.purpurmc.purpur.protocol.core.PurpurCustomPayload payload) {
+                    org.purpurmc.purpur.protocol.core.PurpurProtocolManager.encode(output, payload);
+                    return;
+                }
+                // Purpur end - protocol core
                 this.writeCap(output, value.type(), value);
             }
@@ -45,4 +_,9 @@
             public CustomPacketPayload decode(final B input) {
                 Identifier identifier = input.readIdentifier();
-                return (CustomPacketPayload)this.findCodec(identifier).decode(input);
+                // Purpur start - protocol core
+                var payload = org.purpurmc.purpur.protocol.core.PurpurProtocolManager.decode(identifier, input);
+                return java.util.Objects.requireNonNullElseGet(payload, () -> this.findCodec(identifier).decode(input));
+                // Purpur end - protocol core
             }
