--- a/net/minecraft/world/item/StandingAndWallBlockItem.java
+++ b/net/minecraft/world/item/StandingAndWallBlockItem.java
@@ -26,14 +_,14 @@
 
     @Override
     protected @Nullable BlockState getPlacementState(final BlockPlaceContext context) {
-        BlockState wallState = this.wallBlock.getStateForPlacement(context);
+        BlockState wallState = this.wallBlock.getRealStateForPlacement(context); // Purpur - alternativeBlockPlacement
         BlockState stateForPlacement = null;
         LevelReader level = context.getLevel();
         BlockPos pos = context.getClickedPos();
 
         for (Direction direction : context.getNearestLookingDirections()) {
             if (direction != this.attachmentDirection.getOpposite()) {
-                BlockState possibleState = direction == this.attachmentDirection ? this.getBlock().getStateForPlacement(context) : wallState;
+                BlockState possibleState = direction == this.attachmentDirection ? this.getBlock().getRealStateForPlacement(context) : wallState; // Purpur - alternativeBlockPlacement
                 if (possibleState != null && this.canPlace(level, possibleState, pos)) {
                     stateForPlacement = possibleState;
                     break;
